scripts/release.ts

```diff
@@ -0,0 +1,231 @@
+#!/usr/bin/env -S npx ts-node -P tsconfig.scripts.json
+
+import { execSync } from "child_process";
+
+const SV_COMMAND = "yarn run standard-version --infile ./CHANGELOG.md";
+
+const verboseExec = (cmd: string) => {
+  let result;
+
+  try {
+    result = execSync(cmd, { stdio: "pipe" }).toString("utf-8");
+    console.log(`  ✔ ${cmd}`);
+  } catch (error) {
+    throw new Error(`Command \`${cmd}\` failed:\n\n${error}`);
+  }
+
+  return result;
+};
+
+const checkPrerequisites = () => {
+  try {
+    execSync("git --version", {
+      stdio: "ignore",
+    });
+    execSync("hub --version", {
+      stdio: "ignore",
+    });
+  } catch {
+    throw new Error(`
+  Please install missing dependencies:
+    - https://git-scm.com
+    - https://github.com/github/hub
+`);
+  }
+};
+
+const getNewVersion = (): string => {
+  const svResult = execSync(`${SV_COMMAND} --dry-run`, {
+    stdio: "pipe",
+  }).toString("utf-8");
+
+  let version;
+  const VERSION_MATCH = "bumping version in package.json from (.*) to (.*)";
+  const match = svResult.match(VERSION_MATCH);
+
+  if (match) {
+    version = match[2];
+  } else {
+    throw new Error("Could not get version");
+  }
+
+  return version;
+};
+
+const finalizeRelease = () => {
+  const version = process.argv[3];
+  const pullRequestId = process.argv[4];
+
+  if (version === undefined || pullRequestId === undefined) {
+    printWrongArgsMsg();
+    return;
+  }
```

Comment:
I’d prefer to do command line parsing all in one place and pass the parsed parameters as arguments to functions. The last time we touched the release script there was a bug in the command line parsing that wasn’t visible because the parsing code was spread throughout the file.

In addition we should also throw on the wrong args to avoid that the program continues executing and exits with a non-zero status code.

Reply:
Good point, this was also on my to-do list.

Reply:
50b83c6829b92f2b4cba36656581a8adb41935c6